### DISCUSSION TITLE
Termux: Integrating and harmonising bash-it and ble.sh

### DISCUSSION THREAD

**Discussion by b9Joker108** [Bash-it/bash-it#2250] [Q&A] [answered]
Hello

In Termux, I have just recently first installed ble.sh, followed by bash-it. Bash-it, true to its name, has clobbered my ble.sh. I would really like to integrate and harmonise these two projects locally. I have not yet read the documentation for both projects. This post, is really a call for guidance, for how I may go about this from the bash-it side, and what I should really know, and what I should really, expressly, read and take notice of, in the documentation. I am surprised, as I just sourced my backup .bashrc prior to installing bash_it, but ble.sh was not reinstituted for that shell, which confuses me.

All assistance greatly appreciated.

Cheers
Beauford

**Comment by akinomyoga** [comment, marked as answer]
If you install ble.sh in the default location (which is `~/.local`), you can just turn on ble.sh by running

```bash
$ bash-it enable plugin blesh
```

and restarting the session.

**Reply by b9Joker108** [reply]
This really helped!

**Comment by b9Joker108** [comment]
I actually installed ble.sh with:

```bash
# Quick INSTALL to BASHRC (If this doesn't work, please follow Sec 1.3)

git clone --recursive --depth 1 --shallow-submodules https://github.com/akinomyoga/ble.sh.git
make -C ble.sh install PREFIX=~/.local
echo 'source ~/.local/share/blesh/ble.sh' >> ~/.bashrc
```

And it worked with no obstruction, in Termux. 

I recursively synced ~/ble.sh in archival mode, to a subdirectory of my Bash Dotly dotfiles directory, deleted it from where it was first installed, then symlinked the source of truth, to its original location. Ble.sh worked successfully afterwards. I have not moved, the part of the installation within: `~/.local/share/blesh`.

I note your instruction, and executed:

```bash
→ bash-it enable plugin blesh
blesh enabled with priority 250.
```

This output bodes well. I restarted the bash shell session, as instructed. I note that the prompt has not changed, and is still that of bash-it, but that the ble.sh commandline enhancements, such as highlighting and colours, for example, have been reimplemented.

Prior to installing bash-it, I was trying to append the PID of the current shell session, to the prompt of ble.sh, but had no success. I tried a few different ways. I really would like to have the ble.sh prompt back, instead of that of bash-it, along with the PID customisation. I have followed instructions and created: `.blerc` in my ble.sh dotfiles directory, and symlinked it to: `~/`. But, am unsure how to progress, integrating my customisation of the ble.sh prompt, with ble.sh, in .blerc. Is there a particular section of the documentation I should read?

Any direction you provide, is greatly appreciated.

Thanking you in anticipation
Beauford

**Reply by akinomyoga** [reply]
There seems to be confusion. Ble.sh doesn't set any prompt, so there is no "ble.sh's prompt". I think what you saw before you installed Bash-it was the one configured by the default bashrc provided by the distribution.

If you want to disable Bash-it's *theme* (which defines Bash-it's prompt), you can assign an empty string to `BASH_IT_THEME` in `~/.bashrc` (which is provided by Bash-it).

```bash
# In ~/.bashrc

BASH_IT_THEME=
```

Your original `~/.bashrc` should be backed up as `~/.bashrc.bak`. You can copy the necessary part of `~/.bashrc.bak` back to the new `~/.bashrc` provided by Bash-it.

**Reply by akinomyoga** [reply]
> Ble.sh doesn't set any prompt, so there is no "ble.sh's prompt".

Precisely speaking, there do exist "ble's prompts" such as right prompts, transient prompts, and status line, but those settings are supposed to be put in `~/.blerc`, so Bash-it shouldn't interfere with them.

**Reply by akinomyoga** [reply]
> Prior to installing bash-it, I was trying to append the PID of the current shell session, to the prompt of ble.sh, but had no success.

How did you try to achieve that? It should be simply achieved by something like

```bash
PS1='<some other content>$$\$ '
```

**Reply by b9Joker108** [reply]
No, your succinct code looks nothing like my attempts. I resolved not to include the PID directly in the prompt

**Reply by b9Joker108** [reply]
I have actually included this comment of yours as an annotation, and first population, of my .blerc.

**Comment by b9Joker108** [comment]
Thank you so much for your knowledge, your patience, and your time. Due to you, I have contol of my shell once more.

**Comment by b9Joker108** [comment]
> > Ble.sh doesn't set any prompt, so there is no "ble.sh's prompt".
> 
> Precisely speaking, there do exist "ble's prompts" such as right prompts, transient prompts, and status line, but those settings are supposed to be put in `~/.blerc`, so Bash-it shouldn't interfere with them.
